datalake/src/main/java/com/sequenceiq/datalake/cm/RangerCloudIdentityService.java

```diff
@@ -24,21 +27,87 @@
     @Inject
     private SdxService sdxService;
 
-    private void setAzureCloudIdentityMapping(SdxCluster sdxCluster, Map<String, String> azureUserMapping) {
+    private RangerCloudIdentitySyncStatus sucessfulCloudIdSyncStatus(String message) {
+        RangerCloudIdentitySyncStatus status = new RangerCloudIdentitySyncStatus();
+        status.setState(RangerCloudIdentitySyncState.SUCCESS);
+        status.setStatusReason(message);
+        return status;
+    }
+
+    private RangerCloudIdentitySyncStatus failedRangerCloudIdentitySyncStatus(String message) {
+        RangerCloudIdentitySyncStatus status = new RangerCloudIdentitySyncStatus();
+        status.setState(RangerCloudIdentitySyncState.FAILED);
+        status.setStatusReason(message);
+        return status;
+    }
+
+    private RangerCloudIdentitySyncStatus setAzureCloudIdentityMapping(String envCrn, SdxCluster sdxCluster, Map<String, String> azureUserMapping) {
         String stackCrn = sdxCluster.getStackCrn();
-        LOGGER.info("Updating azure cloud id mappings for datalake stack crn = {}", stackCrn);
+        LOGGER.info("Updating azure cloud id mappings for envCrn = {}, datalake stack crn = {}", envCrn, stackCrn);
         try {
-            clouderaManagerRangerUtil.setAzureCloudIdentityMapping(stackCrn, azureUserMapping);
+            if (!clouderaManagerRangerUtil.isCloudIdMappingSupported(sdxCluster.getStackCrn())) {
+                return sucessfulCloudIdSyncStatus("The datalake does not support cloud identity sync. Sync request is ignored.");
```

Comment:
should we have an "unsupported" status instead of successful? successful seems misleading to me.

Reply:
True, I've added a NOT_APPLICABLE status like you've mentioned.

Reply:
I think we can also get rid of this part if we only make the call from FreeIPA on RAZ enabled environments. I'll look into it as a separate item though.